src/DistributedComputations/distributed_architectures.jl

```diff
@@ -230,11 +230,20 @@ function Distributed(child_architecture = CPU();
     # The rank connectivity _ALWAYS_ wraps around (The cartesian processor "grid" is `Periodic`)
     local_connectivity = RankConnectivity(local_index, ranks) 
 
-    # Assign CUDA device if on GPUs
+
+    # Assign GPUs to each MPI rank
     if child_architecture isa GPU
         local_comm = MPI.Comm_split_type(communicator, MPI.COMM_TYPE_SHARED, local_rank)
         node_rank  = MPI.Comm_rank(local_comm)
-        isnothing(devices) ? device!(node_rank % ndevices()) : device!(devices[node_rank+1]) 
+        if child_architecture == CUDAGPU()
+            device_id = isnothing(devices) ? node_rank % CUDA.ndevices() : devices[node_rank+1]
+            CUDA.device!(device_id)
+        end
+        # MOVE THIS IN EXTENSION
+        # if child_architecture == ROCmGPU()
+        #     device_id = isnothing(devices) ? node_rank % length(AMDGPU.devices()) : devices[node_rank+1]
+        #     AMDGPU.device!(device_id)
+        # end
```

Comment:
will this work? You can also use the `switch_device!` method here instead of explicitly calling `CUDA.device!` and `AMDGPU.device!`